dandiapi/api/tests/test_asset.py

```diff
@@ -282,6 +284,32 @@ def test_asset_rest_delete_not_an_owner(api_client, user, version, asset):
     assert asset in Asset.objects.all()
 
 
+@pytest.mark.django_db
+def test_asset_download(api_client, storage, version, asset):
+    # Pretend like AssetBlob was defined with the given storage
+    AssetBlob.blob.field.storage = storage
+
+    version.assets.add(asset)
+
+    response = api_client.get(
+        f'/api/dandisets/{version.dandiset.identifier}/'
+        f'versions/{version.version}/assets/{asset.asset_id}/download/'
+    )
+
+    assert response.status_code == 302
+
+    download_url = response.get('Location')
+    assert download_url == HTTP_URL_RE
+
+    download = requests.get(download_url)
+    cd_header = download.headers.get('Content-Disposition')
+
+    assert cd_header == f'attachment; filename="{os.path.basename(asset.path)}"'
+
+    with asset.blob.blob.file.open('rb') as reader:
```

Comment:
Just an observation unrelated to pr changes since I lack sufficient knowledge: Eh, we used to have .repo.repo in Datalad - it was always fun to debug and also a good memory excercise .